src/opnsense/scripts/dhcp/get_leases6.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/local/bin/python3
+
+"""
+    Copyright (c) 2023 Deciso B.V.
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+
+    2. Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+
+    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+    --------------------------------------------------------------------------------------
+    list dhcpv6 leases
+"""
+import sys
+sys.path.insert(0, "/usr/local/opnsense/site-python")
+import ujson
+import calendar
+import datetime
+import params
+import time
+
+def parse_date(ymd, hms):
+    result = None
+    dt = '%s %s' % (ymd, hms)
+    try:
+        result = calendar.timegm(datetime.datetime.strptime(dt, "%Y/%m/%d %H:%M:%S;").timetuple())
+    except ValueError:
+        result = None
+
+    return result
+
+def parse_iaaddr_iaprefix(input):
+    """
+    parse either an iaaddr or iaprefix segment. return a tuple
+    containing the type parsed and the corresponding segment
+    """
+    seg_type = input[0].split()[0]
+    segment = dict()
+    segment[seg_type] = input[0].split()[1]
+    for line in input[1:]:
+        parts = line.split()
+        field_name = parts[0]
```

Comment:
not sure if this can happen, but an empty string would crash out.  to prevent this from happening, a construct like the following might help:

```
field_name = parts[0] if len(parts) > 0 else ''
```

Reply:
unlikely, but you know what they say about assumptions..